### PR TITLE
Update some descriptions in sample.conf based on newer README entries

### PR DESCRIPTION
**Pull request by IzzySoft**

### CODE REVIEW

File: sample.conf [hunk 1]
```diff
@@ -118,40 +142,45 @@ PageType	cgi
...
 
-#HTMLPost 	
+#HTMLPost 	<BR CLEAR="all">
 
 # HTMLTail defines the HTML code to insert at the bottom of each
 # HTML document, usually to include a link back to your home
 # page or insert a small graphic.
 
-#HTMLTail 
+#HTMLTail <IMG SRC="msfree.png" ALT="100% Micro$oft free!">
```
Comment (StoneStepsInc) [inline sample.conf:179]: Let's no add disrespectful terms. Microsoft does plenty for the community.

File: sample.conf [hunk 2]
```diff
@@ -476,10 +505,13 @@ HideURL		*.ra
 # to define it's search terms.
 
 SearchEngine	yahoo.com	p=
+SearchEngine	altavista.com	q=
```
Comment (StoneStepsInc) [inline sample.conf:508]: `altavista.com` is gone - it's `search.yahoo.com` now. `msn.com` redirects to `bing.com`. `eureka.com` is just gone.

Unless you are you thinking that these are required for processing historical logs. I hadn't thought of that. Maybe this deserves a comment to explain that while they no longer exist, having them in the configuration might be useful for this purpose.

File: sample.conf [hunk 3]
```diff
@@ -118,40 +142,45 @@ PageType	cgi
...
-# they will be inserted in the output stream in order of apperance.
-# Max string size is 80 characters.  Use multiple lines if you need to.
+# HTMLPost defines HTML code that is placed after the title and "summary
+# period/generated on" lines, just before the initial horizontal rule <hr> tag.
+# Normally this keyword isn't needed, but is provided in case you included a
+# large graphic or some other weird formatting tag in the HTMLBody section that
+# needs to be cleaned up or terminated before the main report section.
 
-#HTMLPost 	
+#HTMLPost 	<BR CLEAR="all">
```
Comment (StoneStepsInc) [inline sample.conf:173]: The `clear` attribute for the `br` element is deprecated in favor of CSS.

File: sample.conf [hunk 4]
```diff
@@ -51,6 +51,30 @@
...
+#Incremental	yes
+
+# IncrementalName allows you to specify the filename for saving the
+# incremental data in.  It is similar to the HistoryName option where the
+# name is relative to the specified output directory, unless an absolute
+# filename is specified.  The default is a file named "webalizer.current"
+# kept in the normal output directory.  If you don't specify "Incremental"
+# as 'yes' then this option has no meaning.
+
+#IncrementalName	webalizer.current
```
[the remark this answers is not in the document]
Comment (IzzySoft) [reply sample.conf:76]: Ah, then I wrongly deducted it's being still there by `Incremental` still being available. Will remove it, thanks!

File: sample.conf [hunk 5]
```diff
@@ -51,6 +51,30 @@
...
+# order to continue processing where it left off.  This mode also causes
+# The Webalizer to scan for and ignore duplicate records (records already
+# processed by a previous run).  See the README file for additional
+# information.  The value may be 'yes' or 'no', with a default of 'yes'.
+# The file 'webalizer.current' is used to store the current state data,
+# and is located in the output directory of the program (unless changed
+# with the IncrementalName option below).  Please read at least the section
+# on Incremental processing in the README file before you enable this option.
+
+#Incremental	yes
```
[the remark this answers is not in the document]
Comment (IzzySoft) [reply sample.conf:67]: Agreed, will remove it again. I just saw it's still mentioned in the Readme, so I left it in. But as it shall be used only in very specific situations by folks knowing well what they're doing, we indeed shouldn't tempt newcomers with it :wink:

Btw: Logrotate is where SSW kicks in on my servers as well.

File: sample.conf [hunk 6]
```diff
@@ -118,40 +142,45 @@ PageType	cgi
 
 #DNSChildren	0
 
-# HTMLPre defines HTML code to insert at the very beginning of the
-# file. Use it for server-side script code, like PHP.
+# HTMLPre allows code to be inserted at the very beginning of the HTML files.
+# Be careful not to include any HTML here, as it is inserted before the
+# <html> tag in the file. Use it for server-side scripting capabilities,
+# such as PHP, to insert scripting files and other directives.
```
[the remark this answers is not in the document]
Comment (IzzySoft) [reply sample.conf:148]: To me it's more clear with the sub-clause. Would you mind having it stay?

File: sample.conf [hunk 7]
```diff
@@ -118,40 +142,45 @@ PageType	cgi
...
-# is 80 characters, so use multiple lines if needed.
+# HTMLHead defines HTML code to insert within the <HEAD></HEAD> block.
+# There is no default. Useful for adding scripts to the HTML page, such as
+# Javascript or php3, or even just for adding a few META tags to the document.
 
 #HTMLHead <META NAME="author" CONTENT="The Webalizer">
 
-# HTMLBody defined the HTML code to be inserted right after the opening
-# <body> tag.
+# HTMLBody defined the HTML code to be inserted right after the start tag of
```
[the remark this answers is not in the document]
Comment (IzzySoft) [reply sample.conf:158]: This was taken straight from your Readme, quote:

> This keyword defines HTML code to be placed immediately after the start tag of the report, just before the title and "summary period/generated on" lines. Keep in mind the placement of this code in relation to the title and other aspects of the web page. A typical use is to add a corporate logo (graphic) in the top right.

So it should be rewritten there as well then. I agree, "immediately after the `<body>` tag is much clearer and hence to prefer. I'll rephrase it to

```html
# HTMLBody is positioned immediately after the <body> tag.
# Keep in mind the placement of this code in relation to the title and other
# aspects of the web page. A typical use is to add a corporate logo (graphic)
# in the top right.
# It is an error to have the <body> tag in any HTMLBody entries.
```

File: sample.conf (hunk 1 above)
Comment (IzzySoft) [reply sample.conf:179]: Oops… That was from one of my configs, not intended to "go public"…

File: sample.conf (hunk 2 above)
Comment (IzzySoft) [reply sample.conf:508]: I have removed them for now, but keep them in my "modular config". If you think they are really helpful for historical reasons, they can always be added back with a proper hint.

File: sample.conf (hunk 3 above)
Comment (IzzySoft) [reply sample.conf:173]: I'll remove this change then as well. Just thought it a "clear" example :wink:

File: sample.conf (hunk 6 above)
[the remark this answers is not in the document]
Comment (StoneStepsInc) [reply sample.conf:133]: it's fine to add detail, but "_insert scripting files_" sounds ambiguous to me in that I don't think there's such construct. Do you mean something like server-side includes? If that's what you are after, it's fine, but let's phrase it the way PHP manual describes it, so it's familiar to people.

File: sample.conf (hunk 7 above)
[the remark this answers is not in the document]
Comment (StoneStepsInc) [reply sample.conf]: Agree, they should be in sync. I will review `README.md` after.

File: sample.conf (hunk 6 above)
[the remark this answers is not in the document]
Comment (IzzySoft) [reply sample.conf:133]: Well, that was taken from an older version of the very same file (not my phrasing). But yeah, if we do a Freud/Jung analysis, it gets weirder the longer one looks at it :zany_face:

> […] Use it for server-side scripting capabilities, such as PHP, to insert calls to script files and other directives.

sounds more convincing IMHO. If you agree, I change it to that.

I must admit I've never used `HTMLPre` as 1) I wasn't sure how *exactly* it is meant, and 2) had no idea what use to put it to. So just to confirm I got it right, can you give a valid example?

```shell
HTMLPre /path/to/foobar.php        # with the script having a she-bang
HTMLPre <?php echo "foobar"; ?>    # inline PHP code; can it be multi-line?
HTMLPre /usr/local/bin/prepend.sh  # a shell script, again with she-bang for the interpreter
HTMLPre fortune                    # an ELF binary spitting out a saying/quote
```

All of them correct – and what they spit out to STDOUT would be included?

[the remark this answers is not in the document]
Comment (StoneStepsInc) [reply sample.conf:133]: This is the structure for all those HTML configuration parameters. Their positioning didn't change from the original Webalizer, but the contents did. The original configuration allowed people to replace `DOCTYPE` and `<body>` with their own tags, which made it impossible to introduce any stable HTML content into reports, so their meaning changed.
```html
HTMLPre
<!DOCTYPE HTML>
<html>
  <head>
    ...
    HTMLHead
  </head>

  <body>
    HTMLBody
    ... page header div ...
    HTMLPost

    ... reports ...

    HTMLTail
    ... page footer div ...
    HTMLEnd
  </body>
</html>
```
Those constructs inside `<body>` allow people to wrap header/footer in their own block tags and style them differently. `HTMLPre` allows people to put any PHP, ASP, JSP, etc content, which typically would be just an include to point to a reusable PHP file, like this:
```php
HTMLPre  <?php include 'xyz.php'; ?>
```
Whatever is included, however, must be processed by the server-side scripting engine and cannot generate any output because having any characters before `DOCTYPE` invalidates HTML. In other words, having PHP `echo` in `HTMLPre` would produce invalid HTML.

If the new wording works for you, sure let's use that.

[the remark this answers is not in the document]
Comment (IzzySoft) [reply sample.conf:133]: Got it, thanks! Especially with that "struct" it got very clear (maybe copy-paste that to the documentation/Readme unless it's already there and I just missed it?). Just to make sure, I'd add parts of what you just wrote:

```html
# HTMLPre allows code to be inserted at the very beginning of the HTML files.
# Be careful not to include any HTML here, as it is inserted before the
# <!DOCTYPE> tag in the file. Use it for server-side scripting capabilities,
# such as running specific PHP code whenever the report is viewed.
# Whatever is included, however, must be processed by the server-side scripting
# engine and cannot generate any output because having any characters before <!DOCTYPE>
# invalidates HTML. In other words, having PHP echo in HTMLPre would produce invalid HTML.

# HTMLPre  <?php include 'xyz.php'; ?>
```

**Review by StoneStepsInc — APPROVED** [review]
Sounds good. Thanks for taking the time to edit.

**Comment by IzzySoft** [comment]
Thanks for reviewing! If it looks fine to you, I'd say it's ready for merging. Should you want me for another run after reworking the Readme, just give me a ping :smiley:
